wwwroot/inc/functions.php

```diff
@@ -2160,7 +2161,9 @@ function markupIPAddrList (&$addrlist)
 			$nallocs++;
 		}
 		$nreserved = ($addrlist[$ip_bin]['reserved'] == 'yes') ? 1 : 0; // only one reservation is possible ever
-		if ($nallocs > 1 && $nallocs != $refc['shared'] || $nallocs && $nreserved)
+		if ($nallocs && $nreserved ||                                                     // reserved cannot be allocated
+                    $nallocs > 1 && $nallocs != $refc['shared'] && $refc['sharedrouter'] == 0 ||  // multiple shared IPs allowed
```

Comment:
could we define the same constraint in a simpler way?
```php
$nallocs > 1 && $nallocs > ($refc['shared'] + $refc['sharedrouter'])
```

Reply:
I considered this, but looks like this way it would be possible to create some "shared" and some "shared router" allocations for the same IPs and it would not be considered a problem. I think, this should not be the case. So, I explicitly check, that all non-conflicting allocations are of the same type.

Reply:
Ah, got it.